### ISSUE TITLE
Make it easier to find what version is the latest for A.B.x branch

### ISSUE THREAD

**Issue by andreif** [python/pythondotorg#1057]
Maybe it's just me, but I find it unnecessary hard to find the latest version in say 3.3.x branch on the downloads page. I have to scan the list, find a release, and then double-check if I maybe missed a later one.

I see few ways how it can be improved:

A. Mark the latest version in the list.
B. Make a separate paragraph with the list of latest versions.
C. Add a UI filter to highlight/show-only releases of chosen branch.
D. Add a UI filter showing only the latest versions of each branch.

The option C seems more functional since it can give an overview for all releases of chosen branch, and the option D seem easiest to implement and use.

Also, many software products have a diagram of version lifetime and it's being frequently requested for Python as well. Such diagram could also tell what the latest release of each branch is.

For example, on [Django's download page](https://www.djangoproject.com/download/) it is very simple to see what the latest version of each branch is (though harder to see the history of each branch).

**Comment by ned-deily** [comment]
I agree that we should have a better overview of releases.  I have had some ideas of how to do that so I'm going to assign this to myself for now.  Thanks for opening this.

**Comment by hugovk** [comment]
I think we can close this now, https://www.python.org/downloads/ has had several improvements.

The "Download" links in this table will take you to the latest patch release for each X.Y:

<img width="1147" height="388" alt="Image" src="https://github.com/user-attachments/assets/62ae4f1b-616e-481a-9ef4-98c6403a6d2f" />

The big release table is sorted by release, rather than date, to make it easier to spot the latest for each X.Y, and also find any given patch for an X.Y:

<img width="1155" height="421" alt="Image" src="https://github.com/user-attachments/assets/35159330-afff-4b8f-b6b4-1c75c2429fe9" />
